python/tests/plugins/test_gen_design_doc_impl.py

```diff
@@ -14,13 +15,13 @@ def test_can_generate_design_doc_with_models(self):
             test_design_doc_file_name = None
```

Comment:
If you can, it would be good to update this test to follow the pattern we set up in the other test (i.e. instead of manually writing the file with the `with open(...)` part, update the test to use `with temporary_test_file(...)` etc.

Reply:
Updated the first test, please let me know if there is anything else required. Thanks Cameron for that final nudge on getting that portion through :)